pkg/cluster/fixmcsuserdata_test.go

```diff
@@ -184,3 +192,76 @@ func TestFixMCSUserData(t *testing.T) {
 		}
 	}
 }
+
+func TestGetUserDataSecretReference(t *testing.T) {
+	for _, td := range []getUserDataSecretReferenceTestData{
```

Comment:
We normally just do this in tests (so that you don't have to define `getUserDataSecretReferenceTestData`:

```suggestion
	for _, td := range []struct {
		name        string
		objectMeta  *metav1.ObjectMeta
		machineSpec *machinev1beta1.MachineSpec
		result      *corev1.SecretReference
		shouldFail  bool
	}{
```